akka-typed/src/main/scala/akka/typed/cluster/ClusterSingleton.scala

```diff
@@ -0,0 +1,185 @@
+
+  def withNoRole(): ClusterSingletonSettings =
+    new ClusterSingletonSettings(None, dataCenter, singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  def withDataCenter(dataCenter: DataCenter): ClusterSingletonSettings =
+    new ClusterSingletonSettings(None, Some(dataCenter), singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  def withNoDataCenter(): ClusterSingletonSettings =
+    new ClusterSingletonSettings(None, None, singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  def withRemovalMargin(removalMargin: FiniteDuration): ClusterSingletonSettings =
+    new ClusterSingletonSettings(role, dataCenter, singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  def withHandoverRetryInterval(handOverRetryInterval: FiniteDuration): ClusterSingletonSettings =
+    new ClusterSingletonSettings(role, dataCenter, singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  def withBufferSize(bufferSize: Int): ClusterSingletonSettings =
+    new ClusterSingletonSettings(role, dataCenter, singletonIdentificationInterval, removalMargin, handOverRetryInterval, bufferSize)
+
+  /**
+   * INTERNAL API:
+   */
+  @InternalApi
+  private[akka] def toManagerSettings(singletonName: String): ClusterSingletonManagerSettings =
+    new ClusterSingletonManagerSettings(singletonName, role, removalMargin, handOverRetryInterval)
+
+  /**
+   * INTERNAL API:
+   */
+  @InternalApi
+  private[akka] def toProxySettings(singletonName: String): ClusterSingletonProxySettings =
+    new ClusterSingletonProxySettings(singletonName, role, singletonIdentificationInterval, bufferSize)
+
+  /**
+   * INTERNAL API:
+   */
+  @InternalApi
+  private[akka] def shouldRunManager(cluster: Cluster): Boolean =
+    (role.isEmpty || cluster.selfMember.roles(role.get)) &&
+      (dataCenter.isEmpty || dataCenter.contains(cluster.selfMember.dataCenter))
+
+}
+
+object ClusterSingleton extends ExtensionId[ClusterSingleton] {
+
+  override def createExtension(system: ActorSystem[_]): ClusterSingleton = new ClusterSingletonImpl(system)
+
+  /**
+   * Java API:
+   */
+  def get(system: ActorSystem[_]): ClusterSingleton = apply(system)
+}
+
+/**
+ * INTERNAL API:
+ */
+@InternalApi
+private[akka] object ClusterSingletonImpl {
+  def managerNameFor(singletonName: String) = s"singletonManager${singletonName}"
+}
+
+/**
+ * INTERNAL API:
+ */
+@InternalApi
+private[akka] final class ClusterSingletonImpl(system: ActorSystem[_]) extends ClusterSingleton {
+  require(system.isInstanceOf[ActorSystemAdapter[_]], "only adapted actor systems can be used for the typed cluster singleton")
+  import ClusterSingletonImpl._
+
+  private lazy val cluster = Cluster(system)
+  private val untypedSystem = ActorSystemAdapter.toUntyped(system).asInstanceOf[ExtendedActorSystem]
+
+  override def spawn[A](
```

Comment:
you removed the other `spawn` methods, but I can't find new `behavior` methods for those?

Reply:
Yeah, I went with @ktoso's "let's see if we can live without them", but maybe that was a bad call. 
(I also couldn't figure out a way to pretend the singleton Props are behaviors)

Reply:
Hmm, might have been a bad call perhaps

Reply:
yeah, that might be impossible. let's start with the simple one only then